### PR TITLE
devops: Cleaned up deploy scripts and added docs

### PR DESCRIPTION
**Pull request by wbreza**
- [x] Re-adds root level .editorconfig to control consistent line endings
- [x] Updates line endings on other files
- [x] Removes old deployment scripts that aren't needed
- [x] Updated deploy scripts to accept named flags
- [x] Adds docs for deploy scripts

### CODE REVIEW

File: api/src/LinkyLink/Controllers/ValidatePageController.cs [hunk 1]
```diff
@@ -1,73 +1,73 @@
...
-                    Detail = ex.Message,
-                    Status = StatusCodes.Status400BadRequest,
-                    Type = "/linkylink/clientissue",
-                };
-                return new BadRequestObjectResult(problemDetails);
-            }
-        }
-    }
-}
+﻿using LinkyLink.Models;
```
Comment (mydiemho) [inline api/src/LinkyLink/Controllers/ValidatePageController.cs]: are these all whitespace changes?

File: api/LinkyLink.sln [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
-		{15A8A501-1B2F-43CB-989E-4DECEAB760B3}.Release|Any CPU.Build.0 = Release|Any CPU
-	EndGlobalSection
-	GlobalSection(SolutionProperties) = preSolution
-		HideSolutionNode = FALSE
-	EndGlobalSection
-	GlobalSection(ExtensibilityGlobals) = postSolution
-		SolutionGuid = {EE88538F-0FA3-425F-BFC0-06E42E66B741}
-	EndGlobalSection
-EndGlobal
+﻿
```
Comment (mydiemho) [inline api/LinkyLink.sln]: should this file get check in? isn't it a build artifact?

File: deployment/arm/deploy.sh [hunk 3]
```diff
@@ -2,17 +2,65 @@
 parent_path=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
```
Comment (mydiemho) [inline deployment/arm/deploy.sh]: add flag `set -e` to fail early

File: deployment/bash/frontdoorazuredeploy.json [hunk 4]
```diff
@@ -1,250 +0,0 @@
-{
```
Comment (mydiemho) [inline deployment/bash/frontdoorazuredeploy.json]: this is encapsulated in the `arm/global.json` correct?

File: deployment/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+- Storage Account (Used for front end static app hosting)
+- App Service (Used for backend API service)
+- App Insights (Used for observability)
+- App Service Plan (Used to host the API app service)
+- API Management (Used to create a consistent API surface for all backend micro services)
+
+## Script (deploy-region.sh)
+The regional deployment script uses a combination of `businessUnit`, `appName`, `env` and `location` to construct a scoped suffix used as a naming convention for the resource group / resources required for this application.
+
+The paramters are passed into the `region.json` ARM template and deployed through the Azure CLI
```
Comment (mydiemho) [inline deployment/docs/README.md]: spelling: paramters

File: frontend/.env.production [hunk 6]
```diff
@@ -1,3 +1,3 @@
-VUE_APP_FRONTEND=https://www.theurlist.com
-VUE_APP_FUNCTION_KEY=SteUyT0o2YSt4JW8k2qZ5ShTBcQYQj1pxE8cO7oTrYI4ps92/53o9w==
-VUE_APP_BACKEND=https://www.theurlist.com
+VUE_APP_FRONTEND=https://www.theurlist.com
```
Comment (mydiemho) [inline frontend/.env.production]: we should remove this file and set the value on deploy (probably relate to the task Jafar is working on

File: frontend/.env.development [hunk 7]
```diff
@@ -1,3 +1,3 @@
-VUE_APP_FRONTEND=http://127.0.0.1:8080
-VUE_APP_FUNCTION_KEY=SteUyT0o2YSt4JW8k2qZ5ShTBcQYQj1pxE8cO7oTrYI4ps92/53o9w==
-VUE_APP_BACKEND=https://www.theurlist.com
+VUE_APP_FRONTEND=http://127.0.0.1:8080
+VUE_APP_FUNCTION_KEY=SteUyT0o2YSt4JW8k2qZ5ShTBcQYQj1pxE8cO7oTrYI4ps92/53o9w==
```
Comment (mydiemho) [inline frontend/.env.development]: I know this value is probably not valid anymore, but should still .gitignore this file, similar to appsettings.development.json in the backend code

File: api/LinkyLink.sln (hunk 2 above)
Comment (wbreza) [reply api/LinkyLink.sln]: This is the visual studio solution file... these are only whitespace changes.

File: deployment/bash/frontdoorazuredeploy.json (hunk 4 above)
Comment (wbreza) [reply deployment/bash/frontdoorazuredeploy.json]: Yes, frontdoor config is in global ARM template.

File: frontend/.env.development (hunk 7 above)
Comment (wbreza) [reply frontend/.env.development]: Will keep this out for not to reduce further complicating this PR.

File: frontend/.env.production (hunk 6 above)
Comment (wbreza) [reply frontend/.env.production]: Will keep this out for not to reduce further complicating this PR.

File: api/src/LinkyLink/Controllers/ValidatePageController.cs (hunk 1 above)
Comment (wbreza) [reply api/src/LinkyLink/Controllers/ValidatePageController.cs]: yes

**Comment by wbreza** [comment]
@michaeldeongreen - Addressed your feedback and added inline docs in the scripts

**Comment by wbreza** [comment]
@mydiemho - Addressed all your feedback and answered questions.

File: frontend/.env.production (hunk 6 above)
Comment (alkofahi) [reply frontend/.env.production]: yes, this is the task that I am taking care of

**Pull request merged**